forge-game/src/main/java/forge/game/cost/CostExile.java

```diff
@@ -257,7 +257,7 @@ else if (type.contains("FromTopGrave")) {
     @Override
     protected Card doPayment(Player payer, SpellAbility ability, Card targetCard, final boolean effect) {
         final Game game = targetCard.getGame();
-        Card newCard = game.getAction().exile(targetCard, null, null);
+        Card newCard = game.getAction().exile(targetCard, ability, null);
```

Comment:
Will passing `ability` here cause issues with the few other cases of TriggerExiled?

"Soulherder" shouldn't care but I wonder about SpellAbility.addForgetOnMovedTrigger

Reply:
yea this seems problematic as it could now cause _Sacred Ground_ to trigger I believe?

Reply:
also the logic is wrong anyway
better check the WhileKeyword stuff that already exists for the Emerge one instead